File: Data Structures & Algorithms/three-integer-sum/submission-1.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        res = []
        nums.sort()
        for k in range(len(nums)-2):
            if k > 0 and nums[k] == nums[k - 1]:  # Skip duplicate values for 'k'
                continue
            i = k+1
            j = len(nums)-1
            while (i < j):
                if nums[k] + nums[i] + nums[j] > 0:
                    j -= 1
                elif nums[k] + nums[i] + nums[j] < 0:
                    i += 1
                else:
                    res.append([nums[i],nums[j],nums[k]])
                    while i < j and nums[i] == nums[i + 1]:
                        i += 1
                    while i < j and nums[j] == nums[j - 1]:
                        j -= 1

                    i += 1
                    j -= 1
        return res

File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-1.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int sum = 0;
        int l = 0;
        int res = nums.size() + 1;

        for (int r = 0; r < nums.size(); r++) {
            sum += nums[r];

            while (sum >= target) {
                res = std::min(res, r - l + 1);
                sum -= nums[l];
                l++;
            }
        }

        if (res == nums.size() + 1) return 0;

        return res;

    }
};